servatrice/src/servatrice_database_interface.cpp

```diff
@@ -953,3 +952,90 @@ QList<ServerInfo_Ban> Servatrice_DatabaseInterface::getUserBanHistory(const QStr
 
     return results;
 }
+
+QList<ServerInfo_ChatMessage> Servatrice_DatabaseInterface::getMessageLogHistory(const QString &user, const QString &ipaddress, const QString &gamename, const QString &gameid, const QString &message, bool &chat, bool &game, bool &room, int &range, int &maxresults)
+{
+
+    QList<ServerInfo_ChatMessage> results;
+    ServerInfo_ChatMessage chatMessage;
+
+    if (!checkSql())
+        return results;
+
+    // BUILD QUERY STRING BASED ON PASSED IN VALUES
+    QString queryString = "SELECT * FROM cockatrice_log WHERE `sender_ip` IS NOT NULL";
+    if (!user.isEmpty())
+        queryString.append(" AND (`sender_name` = :user_name OR `target_name` = :user_name)");
+
+    if (!ipaddress.isEmpty())
+        queryString.append(" AND `sender_ip` = :ip_to_find");
+
+    if (!gameid.isEmpty())
+        queryString.append(" AND (`target_id` = :game_id AND `target_type` = 'game')");
+
+    if (!gamename.isEmpty())
+        queryString.append(" AND (`target_name` = :game_name AND `target_type` = 'game')");
+
+    if (!message.isEmpty())
+        queryString.append(" AND `log_message` LIKE :log_message");
+
+    if (chat || game || room)
+        queryString.append(" AND (");
+
```

Comment:
You can probably nest the following 3 if statements within the `if (chat || game || room)` (for efficiency)